src/plugins/intel_cpu/src/nodes/executors/dnnl/dnnl_matmul_primitive.cpp

```diff
@@ -0,0 +1,351 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "nodes/executors/dnnl/dnnl_matmul_primitive.hpp"
+
+#include <oneapi/dnnl/dnnl_types.h>
+
+#include <algorithm>
+#include <common/primitive_attr.hpp>
+#include <common/primitive_desc_iface.hpp>
+#include <common/primitive_iface.hpp>
+#include <cstdint>
+#include <memory>
+#include <oneapi/dnnl/dnnl.hpp>
+#include <oneapi/dnnl/dnnl_common.hpp>
+
+#include "cpu/x64/cpu_isa_traits.hpp"
+#include "cpu_memory.h"
+#include "cpu_types.h"
+#include "dnnl_extension_utils.h"
+#include "dnnl_postops_composer.h"
+#include "dnnl_utils.hpp"
+#include "memory_desc/cpu_memory_desc.h"
+#include "memory_desc/cpu_memory_desc_utils.h"
+#include "memory_desc/dnnl_memory_desc.h"
+#include "nodes/executors/dnnl/dnnl_shape_agnostic_data.hpp"
+#include "nodes/executors/executor.hpp"
+#include "nodes/executors/fullyconnected_config.hpp"
+#include "nodes/executors/matmul_config.hpp"
+#include "nodes/executors/memory_arguments.hpp"
+#include "utils/debug_capabilities.h"
+
+namespace ov {
+namespace intel_cpu {
+
+using namespace dnnl;
+using namespace ov::element;
+using namespace executor;
+
+// @todo rewrite using hash_builder
+size_t DnnlMatMulPrimitive::Key::hash() const {
+    using namespace dnnl::impl;
+    using namespace dnnl::impl::primitive_hashing;
+
+    size_t seed = 0;
+
+    for (const auto& ptr : {src, wei, bias, dst}) {
+        if (ptr) {
+            seed = hash_combine(seed, get_md_hash(*ptr->getDnnlDesc().get()));
+        }
+    }
+
+    seed = hash_combine(seed, get_attr_hash(*attr.get()));
```

Comment:
Shouldn't we add `weightsNonTransposed` attribute to the key?

Reply:
Actually no, we should not
This one is a bit tricky.
The idea is that creation of dnnl::matmul primitive as a fullyconnected executor does not depend on whether the weights are transposed.
We want to always use MxK KxN
Whether the weights are transposed impacts only a weights preparation / prepacking procedure.
When we start to use dnnl_matmul_primitive for the actual MatMul operation, we will need some adaptation for this logic (to have proper support for transposeA and transposeB and const / non-const second input), but I propose to leave it as is for now.

I have also removed this field from the dnnl_fullyconntected_primitive key
Since we also just always use MxK NxK